api/v1alpha4/machinehealthcheck_webhook.go

```diff
@@ -130,7 +132,9 @@ func (m *MachineHealthCheck) validate(old *MachineHealthCheck) error {
 		)
 	}
 
-	if m.Spec.NodeStartupTimeout != nil && m.Spec.NodeStartupTimeout.Seconds() < minNodeStartupTimeout.Seconds() {
+	if m.Spec.NodeStartupTimeout != nil &&
+		m.Spec.NodeStartupTimeout.Seconds() != disabledNodeStartupTimeout.Seconds() &&
+		m.Spec.NodeStartupTimeout.Seconds() < minNodeStartupTimeout.Seconds() {
```

Comment:
As I understood it, the timeout is meant to prevent users from setting a value that is unreasonably though and would cause their machines to be removed too quick. It originally came from a suggestion on the initial implementation https://github.com/kubernetes-sigs/cluster-api/pull/2250#discussion_r382654941

I think even with users disabling the value, it still makes sense to keep the existing minimum for a real life value